core/blockchain.go

```diff
@@ -1149,6 +1149,12 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 		// If the chain is terminating, stop processing blocks
 		if atomic.LoadInt32(&bc.procInterrupt) == 1 {
 			log.Debug("Premature abort during blocks processing")
+			// QUORUM
+			if bc.chainConfig.Istanbul == nil && bc.chainConfig.Clique == nil {
+				// Only returns an error for raft mode
```

Comment:
If this is `raft` specific, can you review the `if` condition to make sure we target `raft` and `raft` only.
As the current condition may satisfy even for `ethhash` or any future consensus

Reply:
Add `IsQuorum` condition to make sure it is only satisfied for `raft`.

Reply:
Please use `bc.hc.Engine()` to identify consensus algorithm.
For raft, it's `Ethash` with `ModeFullFake` per:
https://github.com/jpmorganchase/quorum/blob/5cb92ceb3ccde7d30464f43df5f869aec3722499/eth/backend.go#L278-L282

Reply:
Changed to use `bc.engine` to identify consensus algorithm. To reduce redundant logic, reuse `bc.GetConsensus` function in `pm.getConsensusAlgorithm`.

Reply:
Reverted to `bc.chainConfig.IsQuorum && bc.chainConfig.Istanbul == nil && bc.chainConfig.Clique == nil`. We have a check here to make sure Quorum only run one of these three consensuses for now. https://github.com/jpmorganchase/quorum/blob/5cb92ceb3ccde7d30464f43df5f869aec3722499/cmd/geth/config.go#L266-L277